server/core/src/atomic_apply_database_operations.cpp

```diff
@@ -19,6 +19,36 @@
 #include <sstream>
 #include <experimental/iterator>
 
+// TEAM DISCUSSION:
+// - Keep it a library (APIs can invoke this and guard this call with permission checks, etc.).
+// - Automated Ingest, NFSRODS, Change Log Listener would use this library.
+// - IDEA: Any API Plugin that invokes this library needs to provide a [cryptographic] access token in order for things to work.
+//   - How do we provide protection around this?
+//     - Could have a special group that can execute this.
+//     - Do we narrow it even more and put the secret in each server_config.json?
+//       - Scoped to a particular server.
+//       - The plugin specific configuration can contain a list of blessed users and groups, etc.
+//         - Example: {"user_1": "token_1", "user_2": "token_2", "group_1": "token_3"}
```

Comment:
so, this is an authenticated user who ALSO has this token?

or just a defined user in the system who uses this particular skeleton key to do ingest things, and might authenticate separately for other, more normal operations...

Reply:
There is no authentication step for this function. This function is not exposed publicly. New RPC APIs will be able to use this for database operations.

Reply:
gotcha.